### PR TITLE
Modified view & initial behavior

### PR DESCRIPTION
**Pull request by AnyaTyu**
Now view - https://yadi.sk/i/_nkHOLf-u4xRF

@tadatuta

### CODE REVIEW

File: server/renderer.js [hunk 1]
```diff
@@ -27,11 +27,11 @@ function render(req, res, data, context) {
 
     }
 
-    if (req.query.hasOwnProperty('bemjson')) return res.send('<pre>' + JSON.stringify(bemjson, null, 4) + '</pre>');
+    if (req && req.query.hasOwnProperty('bemjson')) return res.send('<pre>' + JSON.stringify(bemjson, null, 4) + '</pre>');
 
     const html = BEMHTML.apply(bemjson);
 
-    return res.send(html);
+    return !send ? res.send(html) : html;
```
Comment (skad0) [inline server/renderer.js]: If not send then send?
Maybe:
`return send ? res.send(html) : html`?

Comment (AnyaTyu) [inline server/renderer.js]: @skad0 I think `send` is a bad name of this mod, but I cann't think a better.
This modifier is needed to manage `return`, we give pure html or we do `res.send(html)`

File: common.blocks/workspace/workspace.js [hunk 2]
```diff
@@ -12,8 +12,8 @@ provide(BEMDOM.decl(this.name, {
 
     saveTm: function() {
         const data = JSON.stringify(this._editor.provideData());
-
-        data && $.post('/tm', { data: data })
+        console.log(data);
```
Comment (tadatuta) [inline common.blocks/workspace/workspace.js]: !

File: common.blocks/workspace/workspace.js [hunk 3]
```diff
@@ -34,6 +34,7 @@ provide(BEMDOM.decl(this.name, {
     },
 
     getTM: function() {
+        // TODO:
```
Comment (tadatuta) [inline common.blocks/workspace/workspace.js]: the best comment ever ;)

File: common.blocks/workspace/workspace.js [hunk 4]
```diff
@@ -46,15 +47,13 @@ provide(BEMDOM.decl(this.name, {
...
+        const content = window.segments[index].altTrans;
 
-        $.get(`/alt?segment=${search}&sourceLang=${sourceLang}&targetLang=${targetLang}`)
-            .then((response) => {
-                BEMDOM.replace(this._altTransl.domElem, response);
-                this._altTransl = this.findBlockInside('alternative-translation');
-                this._editorUnit = unit;
-            }).fail((err) => InfoModal.show(err));
+        BEMDOM.replace(this._altTransl.domElem, content);
+        this._altTransl = this.findBlockInside('alternative-translation');
```
Comment (tadatuta) [inline common.blocks/workspace/workspace.js]: altTransl -> altTrans

File: server/renderer.js (hunk 1 above)
Comment (tadatuta) [inline server/renderer.js]: I do not understand this change at all

File: server/helpers/error.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+function onError(req, res, err, url) {
+    console.log('requested url:', url);
+    _logError(err);
```
Comment (AnyaTyu) [inline server/helpers/error.js:5]: err.stack || err

File: server/helpers/error.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+function onError(req, res, err, url) {
+    console.log('requested url:', url);
```
Comment (AnyaTyu) [inline server/helpers/error.js:4]: https://www.npmjs.com/package/debug-http

File: server/helpers/error.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+function onAjaxError(req, res, err) {
+    var errBody = err.response ? err.response.body : '',
+        code = err.statusCode;
+
+    console.log('Ajax url: ', req.path);
+    _logError(err);
+
+    res.status(code);
```
Comment (AnyaTyu) [inline server/helpers/error.js]: чейнить

File: server/helpers/translator.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        return findSegment(trgLang, srcLang, unit.source.content)
+            .then(data => {
+                unit.altTrans = renderer(null, null, data, { block: 'alternative-translation' }, true);
+
+                if (data.length > 0) {
+                    const value = data[0]; // element with a high percentage of matches
+
+                    if (value.source === unit.source.content) {
+                        unit.target.content = value.target; // insert translation in segment's field target
+                        unit.status = true
```
Comment (AnyaTyu) [inline server/helpers/translator.js]: ;

**Pull request merged**